pkg/queryfrontend/response.proto

```diff
@@ -21,15 +21,16 @@ option (gogoproto.goproto_sizecache_all) = false;
 
 
 message ThanosLabelsResponse {
-  string Status = 1 [(gogoproto.jsontag) = "status"];
+  string Status        = 1 [(gogoproto.jsontag) = "status"];
   repeated string Data = 2 [(gogoproto.jsontag) = "data"];
-  string ErrorType = 3 [(gogoproto.jsontag) = "errorType,omitempty"];
-  string Error = 4 [(gogoproto.jsontag) = "error,omitempty"];
+  string ErrorType     = 3 [(gogoproto.jsontag) = "errorType,omitempty"];
+  string Error          = 4 [(gogoproto.jsontag) = "error,omitempty"];
 }
 
 message ThanosSeriesResponse {
-  string Status = 1 [(gogoproto.jsontag) = "status"];
-  repeated thanos.FullCopyLabelSet Data = 2 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "data", (gogoproto.customtype) = "github.com/thanos-io/thanos/pkg/store/labelpb.LabelSet"];
-  string ErrorType = 3 [(gogoproto.jsontag) = "errorType,omitempty"];
-  string Error = 4 [(gogoproto.jsontag) = "error,omitempty"];
+  string Status                 = 1 [(gogoproto.jsontag) = "status"];
+  // TODO(bwplotka): Experiment with ZLabelSet here.
+  repeated thanos.LabelSet Data = 2 [(gogoproto.nullable) = false, (gogoproto.jsontag) = "data"];
```

Comment:
This change from `FullCopyLabelSet` to `thanos.LabelSet` causes the bug of https://github.com/thanos-io/thanos/issues/3465, which returns nil when unmarshalling responses [here](https://github.com/thanos-io/thanos/blob/master/pkg/queryfrontend/labels_codec.go#L227).

However, this pr fixes this by changing `thanos.LabelSet` to  `thanos.ZLabelSet` [here](https://github.com/thanos-io/thanos/commit/f6f622b705ecf522d97a0ad8ac0d58ba525aabde#diff-14c747da74529c9efe346184a3ede2ccc6fb636b464667ea4102f0ddff1e1db9R86), @bwplotka any idea why this change works?